samples/misc/nif/nif.c

```diff
@@ -947,33 +1081,46 @@ BOOL LNPUBLIC NSFNoteCreateTest(DHANDLE hDB)
 	argc, argv - directly from the command line.
 
 	OUTPUTS:
-	szDBFileName, szTimeVariantView, szViewName - data from 
+	szDBFileName, szTimeVariantView, szViewName, szJSON - data from
 	the command line or the user input when prompted.
 
 *************************************************************************/
-void  LNPUBLIC  ProcessArgs (int argNumber, char *szArgVector[], char *szDBFileName, char *szTimeVariantView, char *szViewName)
+void  LNPUBLIC  ProcessArgs (int argNumber, char *szArgVector[], char *szDBFileName, char *szTimeVariantView, char *szViewName, char *szJSON)
 {
-	if (argNumber != 4)
+	if (argNumber != 5)
 	{
-		printf("Enter the name of Database: ");
+		PRINTLOG("Enter the name of Database: ");
 		fflush(stdout);
 		fgets(szDBFileName, STRING_LENGTH-1, stdin);
 		szDBFileName[strlen(szDBFileName) - 1] = '\0';
 
-		printf("Enter the name of Time Variant View: ");
+		PRINTLOG("Enter the name of Time Variant View: ");
 		fflush(stdout);
 		fgets(szTimeVariantView, STRING_LENGTH-1, stdin);
 		szTimeVariantView[strlen(szTimeVariantView) - 1] = '\0';
 
-		printf("Enter the name of view that is not time varying: ");
+		PRINTLOG("Enter the name of view that is not time varying: ");
 		fflush(stdout);
 		fgets(szViewName, STRING_LENGTH-1, stdin);
 		szViewName[strlen(szViewName) - 1] = '\0';
+
+		PRINTLOG("Type \"json\" to print the read entries in JSON format: ");

```

Comment:
Instead of reading as json from enduser, you can ask to enable or disable json output.

Reply:
if we pass empty string as "" It will disables json output. Moreover Since this sample contains other APIs as well this was done.

Reply:
Is it possible to have --json as argument, it more like --help / -h command line argument. Just a suggestion.